platform/linux-generic/odp_rwlock.c

```diff
@@ -33,9 +33,14 @@ void odp_rwlock_read_lock(odp_rwlock_t *rwlock)
 
 int odp_rwlock_read_trylock(odp_rwlock_t *rwlock)
 {
-	uint32_t zero = 0;
+	uint32_t cnt = odp_atomic_load_u32(&rwlock->cnt);
+
+	while (cnt != (uint32_t)-1) {
```

Comment:
Well. It is an interesting point. This `while` waits for other _read_ locks to finish locking in parallel. It does not wait for write lock to be lifted, if there is one. Is it OK from `_trylock()` semantics?